scripts/normalize-by-median.py

```diff
@@ -43,23 +45,13 @@ def batchwise(coll, size):
 # Returns true if the pair of records are properly pairs
 
 
-# pylint: disable=too-many-locals,too-many-branches
-def normalize_by_median(input_filename, outfp, htable, paired, cutoff,
-                        report_fp=None):
+def WithDiagnostics(ifile, batch_size, fp, paired):
+    index = 0
+    global total, discarded
 
-    desired_coverage = cutoff
-    ksize = htable.ksize()
+    for batch in batchwise(screed.open(ifile, parse_description=False),
+                           batch_size):
 
-    # In paired mode we read two records at a time
-    batch_size = 1
-    if paired:
-        batch_size = 2
-
-    index = -1
-    total = 0
-    discarded = 0
-    for index, batch in enumerate(batchwise(screed.open(
-            input_filename, parse_description=False), batch_size)):
         if index > 0 and index % 100000 == 0:
             print >>sys.stderr, '... kept {kept} of {total} or'\
```

Comment:
Talking about `print`, it would be nice to move to the print function syntax:
- add `from __future__ import print_function` to the top of the script
- move `sys.stderr` to the `file` parameter:
```
print('... kept {kept} of {total} or {perc:2}%'
      .format(kept=total - discarded,
              total=total,
              perc=int(100. - discarded / float(total) * 100.)),
      file=sys.stderr)
```
This will help for the Python 3 compatibility.